endpoints/openrtb2/amp_auction_test.go

```diff
@@ -1028,21 +1029,37 @@ func TestAMPSiteExt(t *testing.T) {
 
 // TestBadRequests makes sure we return 400's on bad requests.
 func TestAmpBadRequests(t *testing.T) {
-	dir := "sample-requests/invalid-whole"
+	dir := "sample-requests/invalid-whole/"
 	files, err := os.ReadDir(dir)
 	assert.NoError(t, err, "Failed to read folder: %s", dir)
 
-	badRequests := make(map[string]json.RawMessage, len(files))
+	mockAmpStoredReq := make(map[string]json.RawMessage, len(files))
+	badRequests := make(map[string]testCase, len(files))
 	for index, file := range files {
-		badRequests[strconv.Itoa(100+index)] = readFile(t, "sample-requests/invalid-whole/"+file.Name())
+		filename := file.Name()
+		fileData := readFile(t, dir+filename)
+
+		test, err := parseTestData(fileData, filename)
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		if skipAmpTest(test) {
```

Comment:
So I see in your PR description mentioning a solution being `"have a separate set of invalid-whole sample tests for AMP"`, but it appears you're just skipping over tests that are not suitable for AMP requests with `skipAmpTests`, rather than creating a new folder specifically for these tests?

I'm fine with the current approach, but I'm curious if you could provide more details on this.

Reply:
one approach was to copy all the test files and edit them as required for amp and the other approach was to to the existing files and skip the ones that are not compatible with amp. This one implements the later. These are reasons auction endpoint tests are different than amp.
1. amp does not support app object
2. amp supports only one imp object
3. interstitial is not supported in amp
4. nil request.ext.prebid.cache is initialised by amp endpoint during
5. storedrequest.id is validated during json marshal


Added separate tests for a few scenarios